apps/taxes/src/modules/taxes/tax-prepare-data.ts

```diff
@@ -0,0 +1,37 @@
+import { TaxBaseLineFragment } from "../../../generated/graphql";
+
+export const getDiscountForLine = (
+  line: TaxBaseLineFragment,
+  totalDiscount: number,
+  allLinesTotal: number
+) => {
+  if (totalDiscount === 0 || allLinesTotal === 0) {
+    return 0;
+  }
+  const lineTotalAmount = Number(line.totalPrice.amount);
+  const discountAmount = (lineTotalAmount / allLinesTotal) * totalDiscount;
+  if (discountAmount > lineTotalAmount) {
+    return lineTotalAmount;
+  }
+  return discountAmount;
+};
+
+export const getTaxCodeFromLine = (line: TaxBaseLineFragment): string => {
+  if (line.sourceLine.__typename === "OrderLine") {
+    return (
+      line.sourceLine.variant?.product.metafield ??
+      line.sourceLine.variant?.product.productType.metafield ??
+      ""
+    );
+  }
+
+  return (
+    (line.sourceLine.productVariant.product.metafield ||
+      line.sourceLine.productVariant.product.productType.metafield) ??
+    ""
+  );
+};
+
+export const formatCalculatedAmount = (amount: number) => {
+  return Number(amount.toFixed(2));
+};
```

Comment:
I'd prefer if we dont export loose funfcitons because they don't provide context inline (I need to scroll to source to figure out this is imported from taxes etc)

Better DX would be something like this

```
export const TaxesLineResolver = {
  getTaxCodeFromLine,
  getDiscountForLine
}
```

If you method don't fit the object, it means that it should even be in this file